lib/engine/game/g_1894/game.rb

```diff
@@ -0,0 +1,403 @@
+          paris_tiles.each { |t| t.add_reservation!(plm, 0) }
+        end
+
+        def init_stock_market
+          Engine::StockMarket.new(self.class::MARKET, [],
+                                  multiple_buy_types: self.class::MULTIPLE_BUY_TYPES)
+        end
+
+        def event_late_corporations_available!
+          @log << "-- Event: #{EVENTS_TEXT['late_corporations_available'][0]} --"
+          @corporations.concat(@late_corporations)
+          @late_corporations = []
+        end
+
+        TILE_LAYS = [
+          { lay: true, upgrade: true },
+          { lay: true, upgrade: :not_if_upgraded, cost: 20, cannot_reuse_same_hex: true },
+        ].freeze
+
+        def can_hold_above_corp_limit?(_entity)
+          true
+        end
+
+        def show_game_cert_limit?
+          false
+        end
+
+        def init_round_finished
+          @players.rotate!(@round.entity_index)
+        end
+
+        def revenue_for(route, stops)
+          revenue = super
+          revenue += pc_bonus(route.corporation, stops)
+          revenue += est_le_sud_bonus(route.corporation, stops)
+          revenue
+        end
+
+        def pc_bonus(corp, stops)
+          corp.assigned?('PC') && stops.map(&:hex).find { |hex| hex.assigned?('PC') } ? 10 : 0
+        end
+
+        def est_le_sud_bonus(corp, stops)
+          corp.id == 'Est' && stops.map(&:hex).find { |hex| hex.id == 'I2' } ? 20 : 0
+        end
+
+        def ferry_marker_available?
+          hex_by_id(ENGLAND_FERRY_SUPPLY).tile.icons.any? { |icon| icon.name == FERRY_MARKER_ICON }
+        end
+
+        def ferry_marker?(entity)
+          return false unless entity.corporation?
+
+          ferry_markers(entity).any?
+        end
+
+        def ferry_markers(entity)
+          entity.all_abilities.select { |ability| ability.description == @ferry_marker_ability.description }
+        end
+
+        def connected_to_england?(entity)
+          graph.reachable_hexes(entity).include?(hex_by_id(ENGLAND_HEX))
+        end
+
+        def can_buy_ferry_marker?(entity)
+          return false unless entity.corporation?
+
+          ferry_marker_available? &&
+            !ferry_marker?(entity) &&
+            buying_power(entity) >= FERRY_MARKER_COST &&
+            connected_to_england?(entity)
+        end
+
+        def buy_ferry_marker(entity)
+          return unless can_buy_ferry_marker?(entity)
+
+          entity.spend(FERRY_MARKER_COST, @bank)
+          entity.add_ability(@ferry_marker_ability.dup)
+          @log << "#{entity.name} buys a ferry marker for $#{FERRY_MARKER_COST}"
+
+          tile_icons = hex_by_id(ENGLAND_FERRY_SUPPLY).tile.icons
+          tile_icons.delete_at(tile_icons.find_index { |icon| icon.name == FERRY_MARKER_ICON })
+
+          graph.clear
+        end
+
+        def block_england
+          england = hex_by_id(ENGLAND_HEX).tile.cities.first
+
+          england.instance_variable_set(:@game, self)
```

Comment:
what's going on here...

Reply:
It's the same as block_va_coalfields in 1828. Here it checks whether to offer the possibility to buy a ferry token (the corporation has to be connected to England).